discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -303,61 +305,61 @@ def parse_sol_tx_batch(db, solana_client_manager, tx_sig_batch_records, retries=
     challenge_bus = index_solana_plays.challenge_event_bus
 
     # Process each batch in parallel
-    with db.scoped_session() as session:
-        with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
-            parse_sol_tx_futures = {
-                executor.submit(
-                    parse_sol_play_transaction,
-                    solana_client_manager,
-                    tx_sig,
-                ): tx_sig
-                for tx_sig in tx_sig_batch_records
-            }
-            try:
-                for future in concurrent.futures.as_completed(parse_sol_tx_futures, timeout=45):
-                    # Returns the properties for a Play object to be created in the db
-                    # can be None so check the value exists
-                    result = future.result()
-                    if result:
-                        user_id, track_id, created_at, source, tx_slot, tx_sig = result
-
-                        # Append plays to a list that will be written if all plays are successfully retrieved
-                        # from the rpc pool
-                        plays.append({
-                            "user_id": user_id,
-                            "track_id": track_id,
-                            "created_at": created_at,
-                            "source": source,
+    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
```

Comment:
this just moves the db session init until we're ready to write the plays. great catch @hareeshnagaraj